presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/JdbcConnectorFactory.java

```diff
@@ -83,14 +85,14 @@ public Connector create(String catalogName, Map<String, String> requiredConfig,
         return injector.getInstance(JdbcConnector.class);
     }
 
-    public interface JdbcModuleProvider
+    public interface JdbcModulesProvider
     {
-        Module getModule(String catalogName);
```

Comment:
You can always call `Modules.combine`. In my perception `Module` is a some kind of "collection", so returning a collection of collections does make much sense.

Reply:
It appears that in Modules.combine(A, B) module B fails to inject anything bound by module A. Such problems didn't appear when using List<Module>.

Reply:
Have you tried:
```
import static io.airlift.configuration.ConfigurationAwareModules.combine;
``
?

Reply:
> It appears that in Modules.combine(A, B) module B fails to inject anything bound by module A.

It does not correct, we are using such composition already already in base JDBC connector. Also from what I understand you trying to remove CredentialProviderModule beans from the initialization as there were initialized exactly this way, out of your control.

Reply:
I'll try `ConfigurationAwareModules.combine` next, with regular `Modules.combine' the bindings from one module weren't injected into the other module.